src/pages/RFOX/components/Overview/Stats.tsx

```diff
@@ -1,12 +1,40 @@
 import { Box, Flex, SimpleGrid, Skeleton, Tag } from '@chakra-ui/react'
+import type { AssetId } from '@shapeshiftoss/caip'
+import { bn } from '@shapeshiftoss/chain-adapters'
+import { RFOX_PROXY_CONTRACT_ADDRESS } from 'contracts/constants'
 import { Amount } from 'components/Amount/Amount'
 import { Text } from 'components/Text'
+import { fromBaseUnit } from 'lib/math'
+import { useStakingBalanceOfQuery } from 'pages/RFOX/hooks/useStakingBalanceOfQuery'
+import { selectAssetById, selectMarketDataByAssetIdUserCurrency } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
 
 import { StatItem } from './StatItem'
 
 const gridColumns = { base: 1, md: 2 }
 
-export const Stats: React.FC = () => {
+type StatsProps = {
+  stakingAssetId: AssetId
+}
+
+export const Stats: React.FC<StatsProps> = ({ stakingAssetId }) => {
+  const stakingAsset = useAppSelector(state => selectAssetById(state, stakingAssetId))
+  const stakingAssetMarketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, stakingAssetId),
+  )
+  const { data: contractBalanceOfUserCurrency, isSuccess: isContractBalanceOfUserCurrencySuccess } =
+    useStakingBalanceOfQuery<string>({
+      stakingAssetAccountAddress: RFOX_PROXY_CONTRACT_ADDRESS,
+      stakingAssetId,
+      select: data =>
+        bn(fromBaseUnit(data.toString(), stakingAsset?.precision ?? 0))
+          .times(stakingAssetMarketData.price)
+          .toFixed(2),
+      enabled: Boolean(stakingAssetMarketData),
+    })
+
+  if (!stakingAsset) return null
```

Comment:
Same concerns as https://github.com/shapeshift/web/pull/7142#discussion_r1642860800

Reply:
Discussed this one out in more details, conceptually like the idea of having a context vs. passing things down, though not a now thing